syntax = "proto3";

package plagiarism_detection;

service PingService {
    rpc Ping(PingRequest) returns (PingResponse) {}
}

service AuthService {
    rpc Login(LoginRequest) returns (LoginResponse) {}
    rpc Register(RegisterRequest) returns (RegisterResponse) {}
}

service FileService {
    rpc UploadFile(UploadFileRequest) returns (UploadFileResponse) {}
}

message PingRequest {
}

message PingResponse {
    int32 status = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    int32 status = 1;
    string token = 2;
}

message RegisterRequest {
    string username = 1;
    string password = 2;
}

message RegisterResponse {
    int32 status = 1;
}

message UploadFileRequest {
    oneof data {
        FileMetadata metadata = 1;
        FileChunk chunk = 2;
    }
}

message FileMetadata {
    string token = 1;
    string filename = 2;
    int32 size = 3;
}

message FileChunk {
    bytes data = 1;
}

message UploadFileResponse {
    int32 status = 1;
    optional int32 file_id = 2;
}